### ISSUE TITLE
Mono data downloading got stuck

### ISSUE THREAD

**Issue by eu9ene** [mozilla/firefox-translations-training#505]
The first task was terminated after 3 days:
https://firefox-ci-tc.services.mozilla.com/tasks/Drz5ugggQAi56SUt6HHLPw/runs/6/logs/public/logs/live.log
This one is still in progress
https://firefox-ci-tc.services.mozilla.com/tasks/Drz5ugggQAi56SUt6HHLPw/runs/7/logs/live/public/logs/live.log

This has something to do with the recent refactoring of the mono downloader.

**Comment by eu9ene** [comment]
Either the same issue or something else is still happening. I see this in the logs:

```
[task 2024-05-01T01:23:16.028Z] [download-mono] Dataset: news-crawl_news.2014
[task 2024-05-01T01:23:16.028Z] [download-mono] Language: ru
[task 2024-05-01T01:23:16.028Z] [download-mono] Max Sentences: 200000000
[task 2024-05-01T01:23:16.028Z] [download-mono] Artifacts: /builds/worker/artifacts
[task 2024-05-01T01:23:16.028Z] [download-mono] File Destination: /builds/worker/artifacts/news_2014.ru.zst
[task 2024-05-01T01:23:16.028Z] [download-mono] Downloading WMT newscrawl monolingual data
[task 2024-05-01T01:23:16.028Z] [download-mono] http://data.statmt.org/news-crawl/ru/news.2014.ru.shuffled.deduped.gz
[task 2024-05-01T01:23:16.028Z] [download-mono] URL: http://data.statmt.org/news-crawl/ru/news.2014.ru.shuffled.deduped.gz
[task 2024-05-01T01:23:17.259Z] [downloads] Download size: 734463118 bytes
[task 2024-05-01T01:23:20.062Z] [downloads] 5% downloaded (36724736/734463118 bytes)
[task 2024-05-01T01:23:22.214Z] [downloads] 10% downloaded (73449472/734463118 bytes)
[task 2024-05-01T01:23:24.379Z] [downloads] 15% downloaded (110174208/734463118 bytes)
[task 2024-05-01T01:23:26.543Z] [downloads] 20% downloaded (146898944/734463118 bytes)
[task 2024-05-01T01:23:28.714Z] [downloads] 25% downloaded (183623680/734463118 bytes)
[task 2024-05-01T01:23:30.880Z] [downloads] 30% downloaded (220340224/734463118 bytes)
[task 2024-05-01T01:23:33.064Z] [downloads] 35% downloaded (257064960/734463118 bytes)
[task 2024-05-01T01:23:35.246Z] [downloads] 40% downloaded (293789696/734463118 bytes)
[task 2024-05-01T01:23:37.418Z] [downloads] 45% downloaded (330514432/734463118 bytes)
[task 2024-05-01T01:23:39.604Z] [downloads] 50% downloaded (367239168/734463118 bytes)
[task 2024-05-01T01:23:41.813Z] [downloads] 55% downloaded (403955712/734463118 bytes)
[task 2024-05-01T01:23:44.020Z] [downloads] 60% downloaded (440680448/734463118 bytes)
[task 2024-05-01T01:23:46.217Z] [downloads] 65% downloaded (477405184/734463118 bytes)
[task 2024-05-01T01:23:48.419Z] [downloads] 70% downloaded (514129920/734463118 bytes)
[task 2024-05-01T01:23:50.620Z] [downloads] 75% downloaded (550854656/734463118 bytes)
[task 2024-05-01T01:23:52.824Z] [downloads] 80% downloaded (587571200/734463118 bytes)
[task 2024-05-01T01:23:55.033Z] [downloads] 85% downloaded (624295936/734463118 bytes)
[task 2024-05-01T01:23:57.233Z] [downloads] 90% downloaded (661020672/734463118 bytes)
[task 2024-05-01T01:23:59.437Z] [downloads] 95% downloaded (697745408/734463118 bytes)
[task 2024-05-01T01:24:01.657Z] [downloads] 100% downloaded - Download finished.
[taskcluster:error] Task has been aborted prematurely. Reason: worker-shutdown
[taskcluster 2024-05-01 03:11:55.492Z] === Task Finished ===
[taskcluster 2024-05-01 03:11:55.492Z] Unsuccessful task run with exit code: -1 completed in 7106.904 seconds
```

Example task group:
https://firefox-ci-tc.services.mozilla.com/tasks/groups/O8CichXHQ167GaIGlFDOFQ#dataset-news

cc @gregtatum

**Comment by gregtatum** [comment]
This should be fixed with #530. Let's re-open if it comes up again.

**Comment by gregtatum** [comment]
I didn't read this close enough. It looks like the download completes, but the task never finishes.

**Comment by eu9ene** [comment]
I merged #530 to my branch https://github.com/mozilla/firefox-translations-training/pull/550 so it's maybe something else, but essentially the same issue.

**Comment by gregtatum** [comment]
The script appears to execute to completion. Perhaps I didn't handle closing all of the resources correctly and it's hanging?
